sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/firestore/RpcQosImpl.java

```diff
@@ -204,7 +210,8 @@ public boolean awaitSafeToProceed(Instant instant) throws InterruptedException {
       state.checkActive();
       Duration shouldThrottleRequest = at.shouldThrottleRequest(instant);
       if (shouldThrottleRequest.compareTo(Duration.ZERO) > 0) {
-        logger.info("Delaying request by {}ms", shouldThrottleRequest.getMillis());
+        long throttleRequestMillis = shouldThrottleRequest.getMillis();
+        logger.debug("Delaying request by {}ms", throttleRequestMillis);
```

Comment:
This is when the adaptive throttler backs off, right? I think I'd tend toward an `INFO` log here, and a `due to previous failures` in the message.

Reply:
I dropped it to debug after adding the diagnostic distribution `qos_adaptiveThrottler_throttlingMs` so as not to blow out logs as this scenario can happen thousands of times for a job.

This screenshot shows the average number of times the adaptive throttler throttled grouped by docCount and fieldCount.

![image](https://user-images.githubusercontent.com/1754037/117376687-ad922000-ae9f-11eb-814f-95263cacd598.png)

Reply:
Is `throttlingMs` always reported, or does it hinge on that flag? As long as a user can easily find out what's happening that's fine.